cdap-ui/app/cdap/components/CaskWizards/MicroserviceUpload/MicroserviceQueueEditor.js

```diff
@@ -0,0 +1,184 @@
+    connection: state.general.queues[ownProps.index].properties.connection,
+    sslKeystoreFilePath: state.general.queues[ownProps.index].properties['ssl-keystore-file-path'],
+    sslKeystorePassword: state.general.queues[ownProps.index].properties['ssl-keystore-password'],
+    sslKeystoreKeyPassword: state.general.queues[ownProps.index].properties['ssl-keystore-key-password'],
+    sslKeystoreType: state.general.queues[ownProps.index].properties['ssl-keystore-type'],
+    sslTruststoreFilePath: state.general.queues[ownProps.index].properties['ssl-truststore-file-path'],
+    sslTruststorePassword: state.general.queues[ownProps.index].properties['ssl-truststore-password'],
+    sslTruststoreType: state.general.queues[ownProps.index].properties['ssl-truststore-type']
+  };
+};
+
+const fieldToActionMap = {
+  name: MicroserviceQueueActions.setName,
+  type: MicroserviceQueueActions.setType,
+  namespace: MicroserviceQueueActions.setNamespace,
+  topic: MicroserviceQueueActions.setTopic,
+  region: MicroserviceQueueActions.setRegion,
+  accessKey: MicroserviceQueueActions.setAccessKey,
+  accessId: MicroserviceQueueActions.setAccessId,
+  queueName: MicroserviceQueueActions.setQueueName,
+  endpoint: MicroserviceQueueActions.setEndpoint,
+  connection: MicroserviceQueueActions.setConnection,
+  sslKeystoreFilePath: MicroserviceQueueActions.setSSLKeystoreFilePath,
+  sslKeystorePassword: MicroserviceQueueActions.setSSLKeystorePassword,
+  sslKeystoreKeyPassword: MicroserviceQueueActions.setSSLKeystoreKeyPassword,
+  sslKeystoreType: MicroserviceQueueActions.setSSLKeystoreType,
+  sslTruststoreFilePath: MicroserviceQueueActions.setSSLTruststoreFilePath,
+  sslTruststorePassword: MicroserviceQueueActions.setSSLTruststorePassword,
+  sslTruststoreType: MicroserviceQueueActions.setSSLTruststoreType
+};
+
+const mapDispatchToFieldNameProps = (dispatch, ownProps) => {
+  return {
+    onRemove: () => {
+      dispatch({
+        type: MicroserviceQueueActions.deleteQueue,
+        payload: {index: ownProps.index}
+      });
+    },
+    onAdd: () => {
+      dispatch({
+        type: MicroserviceQueueActions.addQueue,
+        payload: {index: ownProps.index}
+      });
+    },
+    onChange: (fieldProp, e) => {
+      dispatch({
+        type: fieldToActionMap[fieldProp],
+        payload: {
+          index: ownProps.index,
+          [fieldProp]: e.target.value
+        }
+      });
+    }
+  };
+};
+
+let MicroserviceQueueCopy = connect(
+  mapStateToFieldNameProps,
+  mapDispatchToFieldNameProps
+)(MicroserviceQueue);
+
+export default class MicroserviceQueueEditor extends Component {
+  static propTypes = {
+    values: PropTypes.shape({
+      queues: PropTypes.arrayOf(PropTypes.shape({
+        name: PropTypes.string,
+        type: PropTypes.string,
+        properties: PropTypes.object
+      }))
+    }),
+    onChange: PropTypes.func,
+    className: PropTypes.string
+  };
+
+  componentWillMount() {
```

Comment:
For some reason moving the code to `componentDidMount` threw a JS error, so I moved it to `constructor` instead. The docs also recommend using `constructor` instead of `componentWillMount`.